### PR TITLE
Fix text rendering where TextAlignment != Left.

### LINKED ISSUE #1625: Changing TextBlock.Text in UserControl code has no effect in UI
I have a UserControl inside a View. UserControl has a TextBlock, which Text property is bound to code behind property. When property value is updated in code, it does not change in UI, unless some element is repainted (for example ProgressBar Value changes). This issue is present for TextBlock and TextBox. Hovering on TextBox updates the value in UI.

Tried to set value to TextBlock.Text directly, still does not update in UI. Tried Avalonia 0.6.1, 0.6.2-build5276-beta and several nightly versions in between. Still the same behavior.

Here is [repo](https://github.com/Feroks/Avalonia.Repaint) with example. I did not manage to find this issue reported. Am I using UserControls incorrectly?

### PR DESCRIPTION
**Pull request by grokys**
## What does the pull request do?

Fixes rendering of `TextBlock`s where `TextAlignment != Left` as described in #1625.

## What is the current behavior?

If `TextAlignment != Left`, the [bounds passed to `TextNode`](https://github.com/AvaloniaUI/Avalonia/blob/bfdaea8055f7af78622cd959be9ed9222538ef2b/src/Avalonia.Visuals/Rendering/SceneGraph/TextNode.cs#L30) do not accurately reflect the bounds of the text to be drawn, causing changes to the `TextBlock` to not get rendered.

## What is the updated/expected behavior with this PR?

Non-left aligned text is updated correctly by the renderer.

## How was the solution implemented (if it's not obvious)?

To use the correct bounds, this PR changes `FormattedTextImpl.Size` to `Bounds`, which returns a rect that takes into account the text alignment and similarly changes `FormattedText.Measure` to a property `Bounds`.

## Breaking changes

`FormattedText.Measure()` changed to `Bounds` property.

## Fixed issues

Fixes #1625

cc: @Gillibald

### CODE REVIEW

File: src/Avalonia.Visuals/Media/FormattedText.cs [hunk 1]
```diff
@@ -159,13 +159,10 @@ public IEnumerable<Rect> HitTestTextRange(int index, int length)
...
-        /// Gets the size of the text, taking <see cref="Constraint"/> into account.
+        /// Gets the bounds of the text, taking <see cref="Constraint"/> into account.
         /// </summary>
-        /// <returns>The bounds box of the text.</returns>
-        public Size Measure()
-        {
-            return PlatformImpl.Size;
-        }
+        /// <returns>The bounds of the text.</returns>
+        public Rect Measure() => PlatformImpl.Bounds;
```
Comment (Gillibald) [inline src/Avalonia.Visuals/Media/FormattedText.cs]: We will never calculate anything here. Text layouts are immutable. The moment we construct one measures are fixed. I suggest changing this to a property `public Rect Bounds { get; }` or just use `PlatformImpl.Bounds`

Comment (grokys) [reply src/Avalonia.Visuals/Media/FormattedText.cs]: I was going to say "we got that API from WPF", but it seems we didn't! Yes, will change it to a `Bounds` property.

Comment (grokys) [reply src/Avalonia.Visuals/Media/FormattedText.cs]: Done!

**Comment by grokys** [comment]
Going to merge this despite no approvals because @Gillibald has taken a look and it's been 10 days.

**Pull request merged**